### ISSUE TITLE
presto job working in cluster mode, one sql query just on a worker node ,why?

### ISSUE THREAD

**Issue by liyanjie918** [prestodb/presto#5473]
I had deployed a presto cluster environment,a coordinator  node, two worker nodes,the config file follow:

coordinator config
 coordinator=true
node-scheduler.include-coordinator=false
http-server.http.port=8080
query.max-memory=50GB
query.max-memory-per-node=1GB
query.initial-hash-partitions=16
discovery-server.enabled=true
discovery.uri=http://192.168.1.170:8080

worker config:
coordinator=false
http-server.http.port=8080
query.max-memory=50GB
query.max-memory-per-node=1GB
query.initial-hash-partitions=16
discovery.uri=http://192.168.1.170:8080

the connector is mongodb,i exectue query like this:
select * from (select *,row_number() over(partition by 1) as rn from ceshi) where rn>10 and rn <21;

I found this query works on a node,the client informations:
Query 20160620_181859_00031_ey984, FINISHED, 1 node
Splits: 3 total, 1 done (33.33%)
2:05 [5.8M rows, 64B] [46.4K rows/s, 0B/s]

I want to know the query why is not running on all the worker node?

**Comment by blankroom** [comment]
I have the same problem that my presto cluster contains 7 worker nodes but when I execute query, there are only 4 nodes working.

**Comment by akshatnair** [comment]
Do you see all the nodes listed when you execute select * from system.runtime.nodes?

**Comment by blankroom** [comment]
@akshatnair  No, I can't see all the nodes and that exactly confused me because all the worker nodes have the same configurations. I once thought the problem may be that I didn't use uuid as node.id, but after I change it to uuid, it still doesn't work. Now I have no idea what is wrong.

**Comment by blankroom** [comment]
@akshatnair @liyanjie918  Oh, I have just solved my problem. I didn't stop service iptables. After I stop service iptables and ip6tables, the cluster can run correctly.

**Comment by liyanjie918** [comment]
I have already found a solution to this problem, thank you





yj_li（李彦杰）
北京迈迪生科技有限公司
BeiJing MyDC Technology Co., Ltd
北京市海淀区马连洼北路59号亿城国际中心3F0312 
Mobile: 86-15810932095
Email:li_yj@mydc365.com
http://www.mydc365.com
 
 
From: Akshat Nair
Date: 2016-07-13 01:10
To: prestodb/presto
CC: liyanjie918; Author
Subject: Re: [prestodb/presto] presto job working in cluster mode, one sql query just on a worker node ,why? (#5473)
Do you see all the nodes listed when you execute select * from system.runtime.nodes?

**Comment by nezihyigitbasi** [comment]
Closing as it seems like the problem is resolved.

**Comment by rampatel5** [comment]
Hi I am also getting the same issue.
stoped services iptables and ip6tables but do'est work for me.
presto job working in cluster mode, one sql query just on a worker node ,why?

I configured a one node as coordinator and worker and one node as worker only.
while runnning the query it is showing only one node.
checked in the below tables showing one node only which is the coordinator and worker both (main).

select * from system.runtime.nodes;

**Comment by functicons** [comment]
I got the same issue. I had 5 worker nodes, but my query was only executed on 1 node. In my particular case, it was due to I ran all Presto nodes inside of docker containers on different hosts of the cluster. Initially, I used bridge network (the default of Docker), which caused each container got the same IP, e.g., 172.17.0.2. `select * from system.runtime.nodes;` helped me realize the problem. Then I changed my container to use host network, each container will use the host IP and listen on the Presto port (8080 by default), the problem was solved.

I think Presto should add a check, coordinator should disallow different worker nodes register the same IP:port.
